### PR TITLE
Add documentation clarification

### PR DESCRIPTION
**Pull request by XavierGeerinck**
Signed-off-by: Xavier Geerinck <xavier.geerinck@gmail.com>

# Description

Clarify documentation for:
* Client and Server
* Updated Samples
* Added Query State API (For Dapr 1.7.0)

## Issue reference

This works on issues #152 and #159

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: daprdocs/content/en/js-sdk-docs/_index.md [hunk 1]
```diff
@@ -9,15 +9,45 @@ no_list: true
 
 The Dapr JS SDK will allow you to interface with the Dapr process that abstracts several commonly used functionalities such as Service-to-Service invocation, State Management, PubSub, and more.
 
+## Installation
+
+To get started with the Javascript SDK, you can download the Dapr Javascript SDK package from [NPM](https://npmjs.org/package/dapr-client) by running the below:
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/_index.md:14]:
```suggestion
To get started with the Javascript SDK, you can download the Dapr Javascript SDK package from [NPM](https://npmjs.org/package/dapr-client) by running the following:
```

File: daprdocs/content/en/js-sdk-docs/_index.md [hunk 2]
```diff
@@ -9,15 +9,45 @@ no_list: true
...
+
+```bash
+npm install --save dapr-client
+```
+
+## Structure
+
+The Dapr Javascript SDK exists out of 2 major components:
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/_index.md:22]:
```suggestion
The Dapr Javascript SDK contains two major components: 
```

File: daprdocs/content/en/js-sdk-docs/_index.md [hunk 3]
```diff
@@ -9,15 +9,45 @@ no_list: true
...
+```
+
+## Structure
+
+The Dapr Javascript SDK exists out of 2 major components: 
+
+* **DaprServer:** The Dapr Server manages all communication from the Dapr Sidecar to your application
+* **DaprClient:** The Dapr Client manages all communication from your application to the Dapr Sidecar
+
+To achieve communication, you are able to choose between 2 different protocols that are implemented: gRPC or HTTP
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/_index.md:27]:
```suggestion
The above communication can be configured to use either of the gRPC or HTTP protocols.
```

File: daprdocs/content/en/js-sdk-docs/_index.md [hunk 4]
```diff
@@ -9,15 +9,45 @@ no_list: true
...
+## Structure
+
+The Dapr Javascript SDK exists out of 2 major components: 
+
+* **DaprServer:** The Dapr Server manages all communication from the Dapr Sidecar to your application
+* **DaprClient:** The Dapr Client manages all communication from your application to the Dapr Sidecar
+
+To achieve communication, you are able to choose between 2 different protocols that are implemented: gRPC or HTTP 
+
+![](./js-server/dapr-server.jpg)
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/_index.md:29]: nit: alt text for images

File: daprdocs/content/en/js-sdk-docs/_index.md [hunk 5]
```diff
@@ -9,15 +9,45 @@ no_list: true
...
+![](./js-client/dapr-client.jpg)
+
+## Get Started
+
+To help you get started, feel free to check out the resources below:
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/_index.md:34]:
```suggestion
To help you get started, check out the resources below:
```

File: daprdocs/content/en/js-sdk-docs/_index.md [hunk 6]
```diff
@@ -9,15 +9,45 @@ no_list: true
...
 
 <div class="card-deck">
   <div class="card">
     <div class="card-body">
       <h5 class="card-title"><b>Client</b></h5>
-      <p class="card-text">Create a JavaScript client and interact with a Dapr sidecar and other Dapr applications.</p>
+      <p class="card-text">Create a JavaScript client and interact with a Dapr sidecar and other Dapr applications. (e.g. Publishing events, Output Binding support, ...)</p>
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/_index.md:40]:
```suggestion
      <p class="card-text"> Create a JavaScript client and interact with a Dapr sidecar and other Dapr applications. (e.g., publishing events, output binding support, etc.)</p>
```

File: daprdocs/content/en/js-sdk-docs/_index.md [hunk 7]
```diff
@@ -9,15 +9,45 @@ no_list: true
...
       <a href="{{< ref js-client >}}" class="stretched-link"></a>
     </div>
   </div>
+  <div class="card">
+    <div class="card-body">
+      <h5 class="card-title"><b>Server</b></h5>
+      <p class="card-text">Create a JavaScript server and let the Dapr sidecar interact with your application. (e.g. Subscribing to events, Input Binding support, ...) </p>
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/_index.md:47]:
```suggestion
      <p class="card-text"> Create a JavaScript server and let the Dapr sidecar interact with your application. (e.g., subscribing to events, input binding support, etc.) </p>
```

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 8]
```diff
@@ -6,6 +6,10 @@ weight: 500
 description: JavaScript Client SDK for developing Dapr applications
 ---
 
+## Introduction
+
+The Dapr Client will allow you to perform communication with the Dapr Sidecar and get access to its client facing features such as: Publishing Events, Invoking Output Bindings, State Management, Secret Management, and much more.
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:11]:
```suggestion
The Dapr Client allows you to communicate with the Dapr Sidecar and get access to its client facing features such as  Publishing Events, Invoking Output Bindings, State Management, Secret Management, and much more.
```

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 9]
```diff
@@ -14,13 +18,13 @@ description: JavaScript Client SDK for developing Dapr applications
 
 ## Installing and importing Dapr's JS SDK
 
-Install the SDK with npm:
+1. First install the SDK with `npm`:
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:21]:
```suggestion
1. Install the SDK with `npm`:
```

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 10]
```diff
@@ -14,13 +18,13 @@ description: JavaScript Client SDK for developing Dapr applications
...
 
 ```bash
 npm i dapr-client
 ```
 
-Import the libraries:
+2. Then, Import the libraries:
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:27]:
```suggestion
2. Import the libraries:
```

File: daprdocs/content/en/js-sdk-docs/js-client/_index.md [hunk 11]
```diff
@@ -14,13 +18,13 @@ description: JavaScript Client SDK for developing Dapr applications
 
 ## Installing and importing Dapr's JS SDK
 
-Install the SDK with npm:
+1. First install the SDK with `npm`:
 
 ```bash
 npm i dapr-client
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/js-client/_index.md:24]: Should we add `--save`?

File: daprdocs/content/en/js-sdk-docs/_index.md [hunk 12]
```diff
@@ -9,15 +9,45 @@ no_list: true
...
+## Structure
+
+The Dapr Javascript SDK contains two major components: 
+
+* **DaprServer:** The Dapr Server manages all communication from the Dapr Sidecar to your application
+* **DaprClient:** The Dapr Client manages all communication from your application to the Dapr Sidecar
+
+The above communication can be configured to use either of the gRPC or HTTP protocols.
+
+![Dapr Server](./js-server/dapr-server.jpg)
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/_index.md:29]: The images look slightly oversized in the preview, do you want to constraint it to a specific size? https://github.com/dapr/js-sdk/blob/aa14ebf406fb0ac376d6306971503a244011d771/daprdocs/content/en/js-sdk-docs/_index.md

**Review by shubham1172 — APPROVED** [review]
LGTM @XavierGeerinck!

**Comment by amulyavarote** [comment]
Looks good to me as well! @XavierGeerinck

**Review by amulyavarote — APPROVED** [review]

**Pull request merged**